app/services/arcgis_api/token_keeper.rb

```diff
@@ -0,0 +1,234 @@
+module ArcgisApi
+  # Class for retrieve, refresh and manage caching of Arcgis API token.
+  # @attr_reader [connection] The underlying Faraday::Connection
+  # @attr_writer [prefetch_ttl] number of seconds to control timing of requesting a new token
+
+  # The token is saved in cache as a hash
+  # @example cache entry
+  # {
+  #   value: "XXXXXXXXXX",
+  #   sliding_expires_at: xxxxxxx,
+  #   expires_at: xxxxxxxxx,
+  # }
```

Comment:
should we create a Struct class to represent the data stored in the cache and to help document/serialize it?

Reply:
Sure we can, for foreseeable future it's structure wont change, if a struct may offer help on controlling document/serialization, personally I prefer a class other than struct since it offers more flexibility.

Reply:
I was also thinking about the PII struct, due to historical reason, the same_as_address_id not stored as a boolean but a string, this makes the work to convert it to boolean kind of cumbersome, since similar checks(is it a string or boolean) scattered around the code to achieve the conversion. If it's encapsulated, it may be easier for the conversion process.

Reply:
on the second though , struct should be enough in this case, since we dont need any behavior in foreseeable future.